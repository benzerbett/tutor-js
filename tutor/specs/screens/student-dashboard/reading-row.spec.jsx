import React from 'react';
import SnapShot from 'react-test-renderer';
import { cloneDeep } from 'lodash';
import moment from 'moment-timezone';
import chronokinesis from 'chronokinesis';
import Factory from '../../factories';
import ReadingRow from '../../../src/screens/student-dashboard/reading-row';


describe('Reading Row', function() {
  let props;

  beforeEach(function() {
    chronokinesis.travel(new Date('2017-10-14T12:00:00.000Z'));
    moment.tz.setDefault('America/Chicago');
    props = {
      event: Factory.studentDashboardTask({ type: 'homework' }),
      course: Factory.course(),
    };
  });

  it('renders unstarted', function() {
    props.event.complete_exercise_count = 0;
    expect.snapshot(<ReadingRow {...props} />).toMatchSnapshot();
  });

  it('renders in progress', function() {
    props.event.complete_exercise_count = 1;
    expect.snapshot(<ReadingRow {...props} />).toMatchSnapshot();
  });

  it('renders complete', function() {
    props.event.complete = true;
    expect.snapshot(<ReadingRow {...props} />).toMatchSnapshot();
  });

});
