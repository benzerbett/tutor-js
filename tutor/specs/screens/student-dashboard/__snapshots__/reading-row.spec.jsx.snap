// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reading Row matches snapshot 1`] = `
<a
  aria-label="Work on reading: turn-key leverage content"
  className="task row reading workable"
  data-event-id={1}
  href="/course/1/task/1"
  onClick={[Function]}
  tabIndex={0}
>
  <div
    className="column-icon col-sm-1 col-2"
  >
    <i
      aria-label="reading icon"
      className="icon icon-lg icon-reading"
    />
  </div>
  <div
    className="title col-sm-6 col-10"
  >
    turn-key leverage content
  </div>
  <div
    className="feedback col-sm-3 col-5"
  >
    <span>
      Complete
    </span>
  </div>
  <div
    className="due-at col-sm-2 col-5"
  >
    <time>
      Oct 11, 7:00am
    </time>
  </div>
</a>
`;
